isok-broker/src/config.rs

```diff
@@ -1,24 +1,63 @@
 use crate::api::ApiError;
-use crate::message_broker::MessageBrokerError;
+use crate::transport::TransportError;
 use figment::providers::{Format, Yaml};
 use figment::Figment;
+use reqwest::Url;
 use serde::Deserialize;
 use std::collections::HashMap;
 use std::net::{Ipv4Addr, SocketAddr};
 use std::path::PathBuf;
 
 #[derive(Deserialize, Clone)]
 pub struct Config {
-    pub kafka: KafkaConfig,
+    pub transport: Transport,
     pub api: ApiConfig,
 }
 
+#[derive(Deserialize, Clone)]
+#[serde(tag = "type")]
+pub enum Transport {
+    #[serde(alias = "kafka")]
+    Kafka(KafkaConfig),
+    #[serde(alias = "warp10")]
+    Warp10(Warp10Config),
+}
+
 #[derive(Deserialize, Clone)]
 pub struct KafkaConfig {
     pub topic: String,
     pub properties: HashMap<String, String>,
 }
 
+#[derive(Deserialize, Clone)]
+pub struct Warp10Config {
+    /// Valid Warp10 token that is used to write results as metrics
+    pub write_token: String,
+    /// Warp10 service to which the broker will send metrics, in the form of `http://<host>:<port>`
+    pub endpoint: Url,
+    /// Specific labels that need to be added to all metrics generated by the broker
+    #[serde(default)]
+    pub labels: HashMap<String, String>,
+    /// Allow to have an endpoint that is unsafe to use
+    #[serde(default)]
+    pub insecure: bool,
+    /// Maximum number of check results that can be buffered before sending them to Warp10,
+    /// if batch_interval is not reached
+    #[serde(default = "default_batch_size")]
+    pub batch_size: u64,
+    /// Maximum interval to which we send a batch, if batch_size is not reached, in milliseconds
+    #[serde(default = "default_batch_interval")]
+    pub batch_interval: u64,
```

Comment:
I would prefer an unambiguous new type `Milliseconds` that we can get a `core:time::Duration` from instead of this primitive `u64`

Reply:
Will be discussed next week with config part